### PR TITLE
Add AzureAD Terraform provider

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: .azure-pipelines/azure-pipelines-osx.yml [hunk 1]
```diff
@@ -42,6 +42,10 @@ jobs:
       setup_conda_rc ./ ./recipes ./.ci_support/${CONFIG}.yaml
 
       source run_conda_forge_build_setup
+
+      conda install -c defaults conda=4.7.5 --yes --quiet
+      conda config --env --add pinned_packages conda=4.7.5
+
```
Comment (dhirschfeld) [inline .azure-pipelines/azure-pipelines-osx.yml:48]: OSX hack required because of https://github.com/conda/conda/issues/9845

**Comment by dhirschfeld** [comment]
Ping @conda-forge/terraform! If anyone has merge rights I'm keen to get this in.

Happy to revert the OSX hack in the feedstock when conda is fixed.

File: .azure-pipelines/azure-pipelines-osx.yml (hunk 1 above)
Comment (sodre) [reply .azure-pipelines/azure-pipelines-osx.yml:48]: @dhirschfeld, lets just disable the osx build instead. we will fix these things in bulk once we find a fix.

Comment (sodre) [inline .azure-pipelines/azure-pipelines-osx.yml:48]:
```suggestion
```

File: recipes/terraform-provider-azuread/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+go_compiler:
+  - go-nocgo
+cgo_compiler:
+  - go-cgo
+goos:
+  - linux    # [linux]
+  - darwin   # [osx]
+  - windows  # [win]
+goarch:
+  - amd64
```
Comment (sodre) [inline recipes/terraform-provider-azuread/conda_build_config.yaml:10]:
```suggestion
```
I think these are not needed, as they are found in `conda-forge-pinning`

File: recipes/terraform-provider-azuread/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+build:
+  number: 0
+  script:
+    - pushd {{ pkg_src }}
+    - GOBIN=$(go env GOBIN)                     # [not win]
+    - go build -v -o $GOBIN/{{ x4_plugin }} .   # [not win]
+    - go env GOBIN > gobin                      # [win]
+    - set /p GOBIN= < gobin                     # [win]
+    - del gobin                                 # [win]
```
Comment (sodre) [inline recipes/terraform-provider-azuread/meta.yaml:27]:
```suggestion
    - for /f "usebackq tokens=*" %%a in (`go env GOBIN`) do SET "GOBIN= %%a"  # [win]
```

File: recipes/terraform-provider-azuread/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - go build -v -o %GOBIN%\{{ x4_plugin }} .  # [win]
+
```
Comment (sodre) [inline recipes/terraform-provider-azuread/meta.yaml:29]:
```suggestion
  skip: True  # [osx]
  
```
Disable OSX for now

File: .azure-pipelines/azure-pipelines-osx.yml (hunk 1 above)
Comment (sodre) [inline .azure-pipelines/azure-pipelines-osx.yml:48]:
```suggestion
```

Disable the OSX build instead.

**Comment by dhirschfeld** [comment]
I'll wait to see how https://github.com/conda-forge/terraform-provider-aws-feedstock/pull/64 pans out then update to the config there...

**Comment by sodre** [comment]
@dhirschfeld, I was not able to fix the osx build. My suggestion is to just skip it for now. We will do a mass update after the conflicts are settled.

File: recipes/terraform-provider-azuread/meta.yaml (hunk 4 above)
Comment (dhirschfeld) [reply recipes/terraform-provider-azuread/meta.yaml:28]: Added above

**Comment by dhirschfeld** [comment]
I re-added the custom `conda_build_config.yaml` as otherwise it didn't recognise the `goos` env var:
```
Error: Failed to render jinja template in /home/conda/conda-recipes/terraform-provider-azuread/meta.yaml:
'goos' is undefined
```

**Comment by dhirschfeld** [comment]
I think this is good to go now but feel free to push any changes...

**Comment by dhirschfeld** [comment]
Looks like the `MACOSX_DEPLOYMENT_TARGET` trick didn't work:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=145733&view=logs&j=c572e5d5-7ba7-55ce-5575-7ce1d9f7161e&t=fe245b22-2f08-5a76-43b2-e02a18e3fefa&l=123

File: recipes/terraform-provider-azuread/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - for /f "usebackq tokens=*" %%a in (`go env GOBIN`) do SET GOBIN= %%a  # [win]
+    - go build -v -o %GOBIN%\{{ x4_plugin }} .  # [win]
+
+requirements:
+  build:
+    - {{ compiler('go') }} >=1.13
+  run:
+    - terraform >=0.12
+  run_constrained:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.13") }}  # [osx]
```
Comment (sodre) [inline recipes/terraform-provider-azuread/meta.yaml:35]:
```suggestion
```
this is not needed, it is pushed down as a strong [run_export][run_export] from the go compiler when running on OSX.

[run_export]: https://github.com/conda-forge/go-feedstock/blob/5ad419e63a613e2384809dcd040acbf6f251bf99/recipe/meta.yaml#L91-L93

**Comment by dhirschfeld** [comment]
Thanks for getting it over the line @sodre!

**Comment by dhirschfeld** [comment]
Ping @conda-forge/core - it looks like the feedstock wasn't properly created:
https://github.com/conda-forge/terraform-provider-azuread-feedstock

**Comment by beckermr** [comment]
@sodre This PR pushed changes to the CI scripts still. We need to not do that and fix them.

**Comment by sodre** [comment]
@beckermr, sorry about that. I thought it had been cleared.

**Comment by sodre** [comment]
@beckermr, I double checked. If you look at the number of files changed by the PR it shows 1.

I also looked at the master branch, and the .ci_support/osx64.yaml file has not been changed for the past 5 years. Am I missing something else?

**Comment by sodre** [comment]
> @beckermr, I double checked. If you look at the number of files changed by the PR it shows 1.
> 
> I also looked at the master branch, and the .ci_support/osx64.yaml file has not been changed for the past 5 years. Am I missing something else?

I mean to say 5 months ago :)

**Comment by beckermr** [comment]
What is this commit?

https://github.com/conda-forge/staged-recipes/commit/f7e80da34903c9419a0ce1633e52c042cf578d6a

**Comment by beckermr** [comment]
Ohhhh sorry @sodre! I had this wrong. My apologies!

**Comment by dhirschfeld** [comment]
I did a lot of fumbling about on this PR but I think at the end I cleaned it all up?

I probably should have squashed all the wip commits as they seem to have confused the GitHub UI where the files are showing up as changed when they haven't been :(

**Comment by beckermr** [comment]
Yup. I put the settings on this repo to squash and merge only now. :)

**Comment by dhirschfeld** [comment]
> I put the settings on this repo to squash and merge only now. :)

Probably a good idea! :+1:

I haven't seen this happen before. Perhaps it's a GitHub bug?

image
```
(base) ~\code\github\conda-forge\staged-recipes [master ≡]> git log -n1 .\conda_build_config.yaml
commit 8cade54bdab0c69c839d8bd6c5082be79897cf10
Author: Cyrus Harrison <cyrush@llnl.gov>
Date:   Tue Nov 12 09:07:37 2019 -0800

    adjustments to conduit recipe
```
...it's not even consistent as when you click on the file it shows the *correct* commit:
image

Anyway, sorry for the mess 😬

**Pull request merged**